main/src/cgeo/geocaching/connector/gc/GCParser.java

```diff
@@ -1088,13 +1075,36 @@ public int compare(final PocketQuery left, final PocketQuery right) {
 
         while (matcherPocket.find()) {
             final String guid = Html.fromHtml(matcherPocket.group(1)).toString();
+            final String name = Html.fromHtml(matcherPocket.group(2)).toString();
+            int count = 0;
             try {
-                final Integer count = Integer.valueOf(matcherPocket.group(2));
-                downloadablePocketQueries.put(guid, count);
+                count = Integer.parseInt(matcherPocket.group(3));
             } catch (final NumberFormatException e) {
                 Log.w("GCParser.getDownloadablePocketQueries: cannot parse PQ cache count");
                 e.printStackTrace();
             }
+            final String lastGenerationString = matcherPocket.group(4);
+            Date lastGeneration = null;
+            try {
+                lastGeneration = GCLogin.parseGcCustomDate(lastGenerationString);
+            } catch (final ParseException e) {
+                Log.w("GCParser.getDownloadablePocketQueries: cannot parse PQ last generation date");
+                e.printStackTrace();
+            }
+
+            int daysRemaining = 0;
+            try {
+                final String daysRemainingString = matcherPocket.group(6);
+                if (daysRemainingString != null && !"last".equals(daysRemainingString)) {
```

Comment:
Then it would change the meaning. I only can parse an Integer, when the String is not null and not equals "last". If I use (!StringUtils.equals()) it would be true when the String is "last" and when it is null. But I need not null to parse the int. 
Anyway, maybe this code changes when I rework the Regex Pattern.